src/wrapper/jnienv.rs

```diff
@@ -1932,6 +1932,103 @@ impl<'a> JNIEnv<'a> {
         let res = jni_non_void_call!(self.internal, UnregisterNatives, class.into_inner());
         jni_error_code_to_result(res)
     }
+
+    /// Return a pointer to elements of the given Java byte array.
+    ///
+    /// The result is valid until the corresponding release_byte_array_elements() function is
+    /// called. Since the returned array may be a copy of the Java array, changes made to the
+    /// returned array will not necessarily be reflected in the original array until
+    /// release_array_elements() is called.
+    ///
+    /// is_copy is set to JNI_TRUE if a copy is made; or it is set to JNI_FALSE if no copy is made.
+    pub fn get_byte_array_elements(
+        &self,
+        array: jbyteArray,
+        is_copy: &mut jboolean,
+    ) -> Result<*mut jbyte> {
```

Comment:
Perhaps, we can return some wrapper similar to `AutoLocal` instead of the raw pointer?

Reply:
Sure. I was thinking on returning a `Vec` instead of a raw pointer, but I think your suggestion is better / more general.

Reply:
I've implemented the auto-release wrappers for byte and primitive arrays. I've left the original manual / unsafe functions in place, but I'm OK with removing them, if you prefer so.

Was trying to generalize the `AutoArray` containers using generics (as the problem looks highly generalizable). I _almost_ succeeded in doing so, but was stumped by some issues regarding the subjacent JNI API, which has type-specific methods for the `*ArrayElements` family of functions. The tokens in the wrappers need to be type dependant, and I currently don't know how to do that.
So, in the end, I resorted to using two different array containers, 'AutoByteArray`, and `AutoPrimitiveArray` :-/ . Maybe you can suggest better names for these containers, and/or an elegant approach at generalization. I'm OK with iterating over this again, although my Rust knowledge is rapidly reaching its limit here.

Finally, some unit tests for the auto variants are still missing. They will be pushed soon, no later than early next week.

Reply:
Update: unit tests added.